File: app/src/main/java/eu/kanade/mangafeed/event/LibraryMangasEvent.java
package eu.kanade.mangafeed.event;

import android.support.annotation.Nullable;

import java.util.List;
import java.util.Map;

import eu.kanade.mangafeed.data.database.models.Category;
import eu.kanade.mangafeed.data.database.models.Manga;

public class LibraryMangasEvent {

    private final Map<Integer, List<Manga>> mangas;

    public LibraryMangasEvent(Map<Integer, List<Manga>> mangas) {
        this.mangas = mangas;
    }

    public Map<Integer, List<Manga>> getMangas() {
        return mangas;
    }

    @Nullable
    public List<Manga> getMangasForCategory(Category category) {
        return mangas.get(category.id);
    }
}

File: app/src/main/java/eu/kanade/tachiyomi/event/LibraryMangasEvent.kt
package eu.kanade.tachiyomi.event

import eu.kanade.tachiyomi.data.database.models.Category
import eu.kanade.tachiyomi.data.database.models.Manga

class LibraryMangasEvent(val mangas: Map<Int, List<Manga>>) {

    fun getMangasForCategory(category: Category): List<Manga>? {
        return mangas[category.id]
    }
}
